returnsrcvr-checker/src/main/java/org/checkerframework/checker/returnsrcvr/ReturnsRcvrAnnotatedTypeFactory.java

```diff
@@ -29,11 +29,29 @@ public ReturnsRcvrAnnotatedTypeFactory(BaseTypeChecker checker) {
     super(checker);
     THIS_ANNOT = AnnotationBuilder.fromClass(elements, This.class);
 
-    // add supports for frameworks
-    frameworkSupports = new ArrayList<FrameworkSupport>();
-    frameworkSupports.add(new AutoValueSupport());
-    frameworkSupports.add(new LombokSupport());
+    boolean disableAutoValue = false;
```

Comment:
All of this code is shared between the Object Construction Checker and Returns Receiver Checker. Can we factor it out into a common static method in `FrameworkSupport`? That method would take as input the `String` that's currently in the local variable `disabledFrameworkSupports`, and return a list of `FrameworkSupport`s. Then this code duplication isn't necessary.

Reply:
Thanks for the review. The framework supports for object construction checker and returnsrcvr checker are different. Maybe I can factor them out separately and put the methods into the corresponding `FrameworkSupportUtils` class?

Reply:
Good point. I think that your plan makes sense.

That said, the core logic to decide which framework supports to use (i.e. the parsing/toUpperCase code) should be shared.

One design would be to make an enum with the values AUTO_VALUE and LOMBOK, and then return that in the common code. Then, the code in each checker could convert the shared enum into an appropriate list of `FrameworkSupport`s.

Reply:
For the common logic part, I am wondering where should I put it into? I can try to factor it out into a static method in class `ReturnsRcvrChecker`, which is also the place I put the flag name and framework names.

Reply:
@ranmanli I think `FrameworkSupportUtils` inside returns receiver checker is a good place for this code.  You could follow the design @kelloggm suggests and have a nested enum there, and then a static method that takes the `String` from the option and returns an [`EnumSet`](https://docs.oracle.com/javase/7/docs/api/java/util/EnumSet.html) of which frameworks are specified.  Then each checker could go through the set and disable support for the corresponding frameworks.